### PR TITLE
Upgrade Emojibase and Add emoticon variations

### PR DESCRIPTION
**Pull request by langleyd**
## What's in this PR
- Updates emojibase to 15.3.1 so that we can use their utility `generateEmoticonPermutations` to generate variations of plain text emoticons(e.g. with or without nose `:)` `:-)`).
- Adds an explicit `MAX_EMOJI_VERSION_WEB` to filter out newer emojis from inclusion as web is not ready to upgrade yet (it needs to be upgraded in unison with `twemoji` (which is awkward and beyond the scope of this PR)
- It also adds `MAX_EMOJI_VERSION_ANDROID` and `MAX_EMOJI_VERSION_IOS` to the script that generates the `emojibase.json` for those platforms so that they can move forward.

### CODE REVIEW

**Comment by pixlwave** [comment]
Dropping in uninvited to ask an important question!

> Adds an explicit MAX_EMOJI_VERSION to filter out newer emojis from inclusion as web is not ready to upgrade yet

Does this work per platform? Apple supports 15.1 and I'd love to get the 🙂‍↕️ and 🙂‍↔️ reactions on iOS if we could 😇

**Comment by langleyd** [comment]
@pixlwave yea we could probably load it via an `env` variable. Will take a look.

**Comment by pixlwave** [comment]
Amazing, thank you! 🙌

**Comment by langleyd** [comment]
@bmarty We've split out the max emoji version per platform as per the PR description.

Would you like add emoji 15.1 support similar to iOS? (Web needs to stay on 15.0 for now).

File: src/emoji.ts [hunk 1]
```diff
@@ -100,15 +115,14 @@ export const EMOJI: Emoji[] = EMOJIBASE.map((emojiData) => {
...
     // Add mapping from emoticon to Emoji object
-    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-    Array.isArray(emoji.emoticon)
-      ? emoji.emoticon.forEach((x) => EMOTICON_TO_EMOJI.set(x, emoji))
-      : EMOTICON_TO_EMOJI.set(emoji.emoticon, emoji);
+    const emoticons = [emoji.emoticon]
+      //flatten, in case `emoji.emoticon` is an array
+      .flat()
+      .flatMap((x) => generateEmoticonPermutations(x));
+    emoticons.forEach((x) => EMOTICON_TO_EMOJI.set(x, emoji));
```
Comment (florianduros) [inline src/emoji.ts:124]: It's feel wrong to me to mutate a map/array when doing a map another array.

File: src/emoji.ts [hunk 2]
```diff
@@ -16,13 +16,21 @@ limitations under the License.
...
   // We generate a shortcode based on the label if none exist in the dataset
   shortcodes: string[];
 }
 
+const MAX_EMOJI_VERSION_WEB = 15.0;
+const EMOJI_TO_VERSION = new Map<string, number>();
+for (const [versionString, emojis] of Object.entries(VERSIONS)) {
+  const version = parseFloat(versionString);
+  emojis.forEach((emoji) => EMOJI_TO_VERSION.set(emoji, version));
+}
```
Comment (florianduros) [inline src/emoji.ts:32]: A comment is welcomed to explain why we need it

File: src/emoji.ts (hunk 1 above)
Comment (florianduros) [inline src/emoji.ts:124]: It's feel wrong to me to mutate a map/array when doing a map another array.

**Review by pixlwave — APPROVED** [review]
Looks great from the iOS perspective, thanks for the update 👏

File: src/emoji.ts (hunk 1 above)
Comment (langleyd) [reply src/emoji.ts:124]: Yes, true I found that strange also, I wasn't too comfortable making those logic changes before as there wasn't much tests but we have that now :). Will move the logic that isn't related to the map out to a foreach so it's a bit clearer.

**Review by florianduros — APPROVED** [review]

**Comment by langleyd** [comment]
Confirmed with Android team they want to move to 15.1.

**Pull request merged**